acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/hashicorpVault/HashicorpVaultDocker.java

```diff
@@ -0,0 +1,251 @@
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+import com.github.dockerjava.api.model.Ports.Binding;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.command.ExecStartResultCallback;
+import com.github.dockerjava.core.command.PullImageResultCallback;
+import com.github.dockerjava.core.command.WaitContainerResultCallback;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class HashicorpVaultDocker {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String HASHICORP_VAULT_IMAGE = "vault:latest";
+  private static final int DEFAULT_HTTP_PORT = 8200;
+  public static final String vaultToken = "token";
+  private static final String[] CREATE_ETHSIGNER_SIGNING_KEY_SECRET = {
+    "vault",
+    "kv",
+    "put",
+    "secret/ethsignerSigningKey",
+    "value=8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63"
+  };
+  private static final String[] COMMAND_TO_CHECK_VAULT_IS_UP = {"vault", "status"};
+  private static final String LOCALHOST = "localhost";
+  private static final String EXPECTED_FOR_SECRET_CREATION = "created_time";
+  private static final String EXPECTED_FOR_STATUS = "Sealed";
+
+  private final DockerClient docker;
+  private final String vaultContainerId;
+
+  private int port;
+  private String ipAddress;
+  private DefaultDockerClientConfig dockerConfig;
+
+  public HashicorpVaultDocker(final DockerClient docker) {
+    this.docker = docker;
+    pullVaultImage();
+    this.vaultContainerId = createVaultContainer();
+  }
+
+  public void start() {
+    LOG.info("Starting Hashicorp Vault Docker container: {}", vaultContainerId);
+    docker.startContainerCmd(vaultContainerId).exec();
+
+    this.ipAddress = getDockerHostIp();
+    LOG.info("Docker Host IP address: {}", ipAddress);
+
+    LOG.info("Querying for the Docker dynamically allocated vault port number");
+    final InspectContainerResponse containerResponse =
+        docker.inspectContainerCmd(vaultContainerId).exec();
+
+    final Ports ports = containerResponse.getNetworkSettings().getPorts();
+    port = httpRpcPort(ports);
+    LOG.info("Http port for Hashicorp Vault: {}", port);
+  }
+
+  public void createTestData() {
+    LOG.info("creating the secret in vault that contains the private key.");
+    final ExecCreateCmdResponse execCreateCmdResponse =
+        getExecCreateCmdResponse(CREATE_ETHSIGNER_SIGNING_KEY_SECRET);
+    waitFor(
+        60,
+        () ->
+            assertThat(
+                    runCommandInVaultContainer(execCreateCmdResponse, EXPECTED_FOR_SECRET_CREATION))
+                .isTrue());
+    LOG.info("The secret was created successfully.");
+  }
+
+  // TODO: Same thing done in PantheonNode.java . Reuse? Where to put?
```

Comment:
Lose the TODO

Given this method is three lines long, duplication is fine 😁

Reply:
done